apps/nextjs/src/env/schema.mjs

```diff
@@ -13,9 +13,13 @@ export const serverSchema = z.object({
   AZURE_AD_CLIENT_SECRET: z.string(),
   AZURE_AD_TENANT_ID: z.string(),
   NEXTAUTH_SECRET: z.string(),
+  NEXTAUTH_URL: z.preprocess(
+    (str) => process.env.VERCEL_URL ?? str,
```

Comment:
yes, str is the input that the preprocess method will have.

since VERCEL_URL will be used in prod, we don't define the NEXTAUTH_URL explicitly there. the problem with that is that our validator would then fail. so what this does is it lets the VERCEL_URL be used for validation if it exists, else use the str we passed in (which would be the value we assign the NEXTAUTH_URL in our .env)

Reply:
you should only pass the homepage url, it will use the /api/auth/... automatically